driverdataaggregation/README.md

```diff
@@ -0,0 +1,404 @@
+# Data Aggregation Scaling Tests
+
+SmartSim-Scaling currently offers three data aggregation test versions:
+
+ 1. Data Aggregation Standard               (C++ Client and SmartRedis Orchestrator)
+ 2. Data Aggregation Standard Python        (Python Client and SmartRedis Orchestrator)
+ 3. Data Aggregation Standard File System   (Python Client and File System)
+
+## Client Description
+
+The data aggregation scaling test runs two applications. The first application
+is an MPI application that produces datasets that are added to an aggregation list.
+In this producer application, each MPI rank has a single-threaded client.  The second
+application is a consumer application. This application consumes the aggregation
+lists that are produced by the first application.  The consumer application
+can be configured to use multiple threads for data aggregation.  The producer and consumer
+applications are running at the same time, but the producer application waits for the
+consumer application to finish an aggregation list before starting to produce
+the next aggregation list.
+
+By default, the clients in the producer application perform an adjustable 100 executions of the following command:
+
+  1) ``append_to_list`` (add dataset to the aggregation list)
+
+Note that the client on rank 0 of the producer application performs a ``get_list_length()``
+function invocation prior to an ``MPI_BARRIER`` in order to only produce the next aggregation
+list after the previous aggregation list was consumed by the consumer application.
+
+There is only a single MPI rank for the consumer application, which means there is only
+one SmartRedis client active for the consumer application.  The consumer application client
+invokes the following SmartRedis commands:
+
+  1) ``poll_list_length``         (check when the next aggregation list is ready)
+  2) ``get_datasets_from_list``   (retrieve the data from the aggregation list)
+
+The input parameters to the test are used to generate permutations
+of tests with varying configurations.
+
+## Why Standard Deployement for Data Aggregation vs Colocated?
+
+SmartSim-Scaling supports standard deployment for all three data aggregation tests. For standard deployement, the Orchestrator database and the application will be launched on different client nodes. To perform data aggregation, each thread has to be assigned to a single shard of a database hosted on a client node. Hence, multiple client nodes are needed for sharded database support. 
```

Comment:
Multiple client nodes are not needed for sharded database support. You need multiple database nodes. For dataset aggregation, each client will create a new thread to try to grab a dataset from multiple database shards. Reword this paragraph